src/status_im/contexts/profile/settings/events.cljs

```diff
@@ -158,3 +158,9 @@
           [{:method     "settings_mnemonicWasShown"
             :on-success #(log/debug "mnemonic was marked as shown")
             :on-error   #(log/error "mnemonic was not marked as shown" %)}]]]}))
+
+(rf/reg-event-fx :profile.settings/update-currency
+ (fn [_ [currency]]
+   {:fx [[:dispatch [:profile.settings/profile-update :currency currency]]
+         [:dispatch [:wallet/get-wallet-token-for-all-accounts]]
```

Comment:
We also need to fetch the tokens again (for the market values) on receiving new currency value from the signal

https://github.com/status-im/status-mobile/blob/43651ef0d0020eeea829bad437a6769427225cfe/src/legacy/status_im/multiaccounts/update/core.cljs#L83